<launch>
	<arg name="base_frame" default="robot_0/base_link" />
	<!-- <arg name="odom_frame" default="robot_0/odom_combined" /> -->
	<arg name="odom_frame" default="robot_0/odom" />
	<arg name="map_frame" default="map" />
	<arg name="use_deadman" default="false" />
	<arg name="person_frame" default="person_raw" />
	<param name="map_name" value="real_2"/>
    <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
		<remap from="scan" to="robot_0/base_scan_1"/>
		<param name="odom_frame" value="$(arg odom_frame)"/>
		<param name="base_frame" value="$(arg base_frame)"/>
		<param name="map_frame" value="map"/>
	    <param name="map_update_interval" value="0.025"/>
	    <param name="particles" value="20" />
	    <param name="delta" value="0.05"/>
	    <param name="maxRange" value="7"/>
	    <param name="xmin" value="-10"/>
	    <param name="ymin" value="-10"/>
	    <param name="xmax" value="10"/>
	    <param name="ymax" value="10"/>
	    <rosparam command="load" file="$(find robot_describe)/config_nav/mapper_params.yaml" />
	</node>

    <node pkg="rviz" type="rviz" name="rviz" output="screen" >
		<!--<remap from="/move_base_simple/goal" to="/move_base_flex/current_goal"/>-->
	</node>

		<node pkg="stage_ros" type="stageros" name="stageros_node" args="$(find robot_describe)/worlds/$(param map_name).world">
		<remap from="/robot_0/blob" to="/robot_0/blob_0" />
		<remap from="/robot_0/cmd_vel" to="/cmd_vel" />
	</node>


</launch>